actions/push-to-gcs/README.md

```diff
@@ -0,0 +1,84 @@
+
+```yaml
+name: Upload Files to GCS
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  ENVIRONMENT: "dev"
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  upload-to-gcs:
+    name: upload
+    runs-on: ubuntu-x64-small
+    steps:
+      - uses: actions/checkout@v4
+
+        # upload a file to the root of a bucket
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-yaml-to-root
+        with:
+          path: .github/workflows/upload-files-to-gcs.yaml
+
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-Dockerfile-to-root
+        with:
+          path: Dockerfile
+
+        # upload a file to a folder in the bucket
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-yaml-to-some-path
+        with:
+          path: .github/workflows/upload-files-to-gcs.yaml
+          bucket_path: some-path/
+
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-Dockerfile-to-some-path
+        with:
+          path: Dockerfile
+          bucket_path: some-path
+
+        # upload .yml files ./docs to bucket/docs
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-all-yml-docs
+        with:
+          path: docs
+          glob: "*.yml"
+
+        # upload .yml files from docs to bucket/this-folder/docs
+      - uses: grafana/shared-workflows/actions/push-to-gcs@main
+        name: upload-all-yml-docs
+        with:
+          path: docs
+          glob: "*.yml"
+          bucket_path: this-folder
+```
+
+## Inputs
+
+| Name          | Type   | Description                                                                                                                                                                                                                                                                                                          |
+| ------------- | ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `path`        | String | (Required) Path to the object(s) to upload. Can be a path to file to upload 1 file, or a folder path to upload that folder and its contents. Can be used in conjunction with the `glob` option. Valid examples include `thing.txt` and `path/to/thing.txt`. Valid examples when also using `glob` include `path/to`. |
+| `bucket`      | String | Name of bucket to upload to. (Default: `grafanalabs-${repository.id}-${environment}`)                                                                                                                                                                                                                                |
```

Comment:
Hm I thought of that as well during the GAR work - I think for bucket names it makes more sense to just use names instead of ids. What I want to say by that is that for example we use `github-<repoID>-<env>@blah` for service accounts, but service accounts are something that won't bother us at all or will only bother us in some extreme/edge cases. In contrast, there will be times when we would like to manually check something from a bucket asap, and it would be more obvious to have something like `grafanalabs-grafana-enterprise-dev` rather than `grafanalabs-123456789-dev`. It's for practical reasons mostly. But still - not strongly opinionated. We can always revisit if discovery is hard!

Reply:
Yeah, I know, I went this route because if you change the name of a GH repo you also have to recreate the bucket. This avoids that; but at this point both you and Iain have requested it... so it might be time to redo it.

I agree the user experience is worse with this route.

Reply:
This has been completed.